code/modules/mob/living/carbon/damage_procs.dm

```diff
@@ -84,9 +84,9 @@
 	if(!forced && HAS_TRAIT(src, TRAIT_TOXINLOVER)) //damage becomes healing and healing becomes damage
 		amount = -amount
 		if(amount > 0)
-			blood_volume -= 5*amount
+			blood_volume = max(blood_volume - (5*amount), 0)
 		else
-			blood_volume -= amount
+			blood_volume = max(blood_volume - amount, 0)
 	if(HAS_TRAIT(src, TRAIT_TOXIMMUNE)) //Prevents toxin damage, but not healing
 		amount = min(amount, 0)
```

Comment:
would it be out of scope for this PR to move a copy of this bit of code here to the spot between amount = -amount and if(amount > 0)

this would fix the bug that causes people with both TRAIT_TOXINLOVER and TRAIT_TOXIMMUNE to still suffer large amounts of blood loss from tox healing attempts, even though they don't take any actual tox damage from it.

Reply:
Sure, why not.